src/sysext/sysext.c

```diff
@@ -689,186 +691,384 @@ static int work_dir_for_hierarchy(const char *hierarchy, const char *resolved_up
         return 0;
 }
 
-static int use_hierarchy_as_lowest_layer(
-                const char *hierarchy,
-                const char *maybe_resolved_upper_dir,
-                char** ret_hierarchy_as_lowest_layer) {
-        _cleanup_free_ char *resolved_hierarchy = NULL;
-        int r;
+struct overlayfs_paths {
```

Comment:
while we don't strictly follow this rule, in most cases we'd name this in CamelCase and would typedef it to itself, i.e. `typedef struct OverlayFSPaths { … } OverlayFSPaths`

Reply:
Yeah, I was confused about it when I wrote it. I thought that public types ought to use CamelCase, whereas the internal symbols - lower_case. Will rename.